### PR TITLE
Feat/add artist clutering

### PR DESCRIPTION
**Pull request by lmontier**
# DS PR

## Describe your changes

- This PR modifies the link_artists.py dag and the artist_linkage microservice
- This PR allows to match similar artist_names and give them an artist_nickname

## Jira ticket number and/or notion link

https://www.notion.so/passcultureapp/Implem-ETQDS-quand-je-lance-le-DAG-match_artists-je-vois-sur-la-table-linked_artists-la-liste-des-9ba39daed4464417a092a61014fc031d?p=6a8a879c6d044b99afac55e56a9fac3d&pm=s

### Type of change
- [ ] hotfix (non-breaking change which fixes an issue)
- [ ] New model
- [ ] Training
- [x] New features
- [ ] Bug Fix
- [ ] Code Refacto
- [ ] Performance Improvements
- [ ] Test
- [ ] CI
- [ ] Config

      
### Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] My code passes CI/CD tests
- [ ] I updated README.md
- [x] I have updated the dag
- [ ] If my changes concern incremental table, I have altered their schema to accomodate with field's creation/deletion
- [ ] I have documented the corresponding [notion page](https://www.notion.so/passcultureapp/Team-Data-engineering-Data-science-22ab0eb5ddf34dc2a854d9f0e596e91b)
- [ ] I have made corresponding changes to the [tables documentation](https://www.notion.so/passcultureapp/Documentation-Tables-175a397a8e854ff4a55ae4f3620dbe3b)
- [ ] I have made corresponding changes to the [fields glossary](https://www.notion.so/passcultureapp/854a436a8f1541e1b6ec2a65f8bab600?v=798024ba90404b139e5a17407a3bc604)
- [x] I will create a review on slack and ensure to specify the duration of the review task: short (<10min), medium (<30min), long (>30min)

### Added tests?
- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help
- [ ] ⏰ no, but I created a ticket

### CODE REVIEW

File: jobs/ml_jobs/artist_linkage/cluster.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+SCORE_MULTIPLIER = (
+    255
+    if DTYPE_DISTANCE_MATRIX == np.uint8
+    else 65535
+    if DTYPE_DISTANCE_MATRIX == np.uint16
+    else 1
+)
+DISTANCE_METRIC = rapidfuzz.distance.OSA.normalized_distance
+CLUSTERING_THRESHOLD = 0.1
+
```
Comment (dcuesta-pass) [inline jobs/ml_jobs/artist_linkage/cluster.py:31]: Chaud pour avoir une brief sur ces valeurs :)

File: jobs/ml_jobs/artist_linkage/cluster.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+            min_samples=2,
+            metric="precomputed",
+        )
+        clustering.fit(complete_sparse_matrix)
+        clusters = clustering.labels_
+
+        clusters_df_list.append(
+            pd.DataFrame({"artist": artists_list})
+            .assign(
+                artist_encoding=lambda df: df.artist.map(jellyfish.metaphone),
```
Comment (dcuesta-pass) [inline jobs/ml_jobs/artist_linkage/cluster.py:107]: Idem , preneur d'une explication sur l'artiste_encoding

File: jobs/ml_jobs/artist_linkage/notebooks/st_explore_and_clusterize_data.py
Comment (dcuesta-pass) [inline jobs/ml_jobs/artist_linkage/notebooks/st_explore_and_clusterize_data.py:1]: Les notebooks on a plutôt l'habitude de les mettre sur l'instance dédié. On a aussi un repo dédié (pas utilisé) qu'on pourrait réanimé

File: jobs/ml_jobs/artist_linkage/cluster.py (hunk 1 above)
Comment (lmontier) [reply jobs/ml_jobs/artist_linkage/cluster.py:23]: du coup j'ai ajouté une fonction qui calcule le score multiplier à partir d'un type. Et j'ai mis une docstring.

File: jobs/ml_jobs/artist_linkage/cluster.py (hunk 2 above)
Comment (lmontier) [reply jobs/ml_jobs/artist_linkage/cluster.py]: Vu ensemble l'artist_encoding. Néanmoins vu que je ne m'en sers pas pour l'instant, je l'ai supprimé du code.

File: jobs/ml_jobs/artist_linkage/notebooks/st_explore_and_clusterize_data.py
Comment (lmontier) [reply jobs/ml_jobs/artist_linkage/notebooks/st_explore_and_clusterize_data.py:1]: vu ensemble : renommé en streamlits

**Review by dcuesta-pass — APPROVED** [review]

**Pull request merged**